hydra-node/test/Hydra/Chain/Direct/Contract/Abort.hs

```diff
@@ -168,32 +168,28 @@ genAbortMutation (tx, utxo) =
     , SomeMutation UseInputFromOtherHead <$> do
```

Comment:
This mutation is older than this pull request.

As we did not check for the the _headId_ before, I would have expected this mutation `UseInputFromOtherHead` to fail on master but it does not so I guess the test did not cover some case that this P.R. is fixing.

So now I expect this mutation to make the test succeed inside the P.R. because of the fix brought by the P.R.

It is my understanding that this P.R. improves this mutation to cover the case that was not covered before but it's not clear to me how just looking at the changes.

Could you explain in a word. or two what was missing in the previous version of the mutation?

Reply:
Previously we had a check in the head validator that detects if the ST token is present but we were using just the token name `HydraHeadV1` for this check and we never checked PolicyId/CurrencySymbol of the asset which is what the gap is about. We used to grab whatever CS we got for the token with a matching name and then use that to check the signatures for example. The abort test you are mentioning was properly altering the policy id of the token and we were checking if all the PTs contain the same CS  but we never checked if the CS we get is the one that belongs to that specific head.